Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,442 @@
+void OnePedalDriveDecider(void);
+void AccelerateCruiseHandler(void);
+void AccelerateCruiseDecider(void);
+
+// State Names
+typedef enum{
+    NORMAL_DRIVE,
+    RECORD_VELOCITY,
+    POWERED_CRUISE,
+    COASTING_CRUISE,
+    BRAKE_STATE,
+    ONEPEDAL_DRIVE,
+    ACCELERATE_CRUISE
+} TritiumStateName_t;
+
+// State Struct for FSM
+typedef struct TritiumState{
+    void (*stateHandler)(void);
+    void (*stateDecider)(void);
+} TritiumState_t;
+
+// FSM
+static const TritiumState_t FSM[9] = {
+    {&NormalDriveHandler, &NormalDriveDecider},
+    {&RecordVelocityHandler, &RecordVelocityDecider},
+    {&PoweredCruiseHandler, &PoweredCruiseDecider},
+    {&CoastingCruiseHandler, &CoastingCruiseDecider},
+    {&BrakeHandler, &BrakeDecider},
+    {&OnePedalDriveHandler, &OnePedalDriveDecider},
+    {&AccelerateCruiseHandler, &AccelerateCruiseDecider}
+};
+
+// Current state
+static TritiumState_t state;
+
+/**
+ * @brief Dumps info to UART during testing
+*/
+#ifdef __TEST_SENDTRITIUM
+void dumpInfo(){
+    printf("-------------------\n\r");
+    printf("State: %d\n\r", state);
+    printf("cruiseEnable: %d\n\r", cruiseEnable);
+    printf("cruiseSet: %d\n\r", cruiseSet);
+    printf("regenToggle: %d\n\r", regenToggle);
+    printf("brakePedalPercent: %d\n\r", brakePedalPercent);
+    printf("accelPedalPercent: %d\n\r", accelPedalPercent);
+    printf("reverseSwitch: %d\n\r", reverseSwitch);
+    printf("forwardSwitch: %d\n\r", forwardSwitch);
+    printf("currentSetpoint: %f\n\r", currentSetpoint);
+    printf("velocitySetpoint: %f\n\r", velocitySetpoint);
+    printf("velocityObserved: %f\n\r", velocityObserved);
+    printf("-------------------\n\r");
+}
+#endif
+
+/**
+ * @brief Reads inputs from the system
+*/
+void readInputs(){
+    #ifndef __TEST_SENDTRITIUM
+    Minion_Error_t err;
+
+    brakePedalPercent = Pedals_Read(BRAKE);
+    accelPedalPercent = Pedals_Read(ACCELERATOR);
+    
+    // UPDATE BUTTONS
+    if(Minion_Read_Input(REGEN_SW, &err) && regenCounter < DEBOUNCE_PERIOD){regenCounter++;}
+    else if(regenCounter > 0){regenCounter--;}
+
+    if(Minion_Read_Input(CRUZ_EN, &err) && cruiseEnableCounter < DEBOUNCE_PERIOD){cruiseEnableCounter++;}
+    else if(cruiseEnableCounter > 0){cruiseEnableCounter--;}
+
+    if(Minion_Read_Input(CRUZ_ST, &err) && cruiseSetCounter < DEBOUNCE_PERIOD){cruiseSetCounter++;}
+    else if(cruiseSetCounter > 0){cruiseSetCounter--;}
+    
+    // UPDATE GEARS
+    bool forwardSwitch = Minion_Read_Input(FOR_SW, &err);
+    bool reverseSwitch = Minion_Read_Input(REV_SW, &err);
+    forwardGear = (forwardSwitch && !reverseSwitch);
+    reverseGear = (!forwardSwitch && reverseSwitch);
+    neutralGear = (!forwardSwitch && !reverseSwitch);
+
+    if(forwardSwitch){
+        cruiseSet = Minion_Read_Input(CRUZ_ST, &err);
+    }
+    else{
+        cruiseEnable = false;
+        cruiseSet = false;
+        regenToggle = false;
+    }
+
+    if(forwardGear ^ reverseGear ^ neutralGear == false)
```

Comment:
I think this is trying to say that if more than 1 is set or if none are set, we have a fault. However, I don't think this will work because it will sequentially evaluate them, and it's not a true 3-way XOR. I think a more failsafe way to do it is the following:
```C
uint8_t gearFault = (uint8_t) forwardGear + (uint8_t) reverseGear + (uint8_t) neutralGear;

if(gearFault!=1){
//whatever fault routine you're going to build
}

```

Reply:
Also, just as a note, if a gear cannot be determined, neutral gear is the safest.